public/app/features/query/components/QueryEditorRows.tsx

```diff
@@ -66,14 +66,14 @@ export class QueryEditorRows extends PureComponent<Props> {
   onReplaceQuery(query: DataQuery, index: number) {
     const { queries, onQueriesChange, onUpdateDatasources, dsSettings } = this.props;
 
-    // Replace old query with new query
+    // Replace old query with new query, preserving the original refId
```

Comment:
This fix is required, if we don't preserve the `refId` the new query could have the same `refId` as an existing query and refIds should be unique